image/app/src/create/mod.rs

```diff
@@ -27,6 +28,46 @@ use std::path::PathBuf;
 use caliptra_image_elf::ElfExecutable;
 use config::{OwnerKeyConfig, VendorKeyConfig};
 
+use chrono::NaiveDate;
+
+///
+/// This function takes the string as the input
+/// and extracts year, month and day from the string
+/// and then performs some basic validity checks
+///
+fn check_date(from_date: &str, to_date: &str) -> anyhow::Result<bool> {
+    let time_fmt = "YYYYMMDDHHMMSS";
+
+    let current_date = chrono::Utc::now().date_naive();
+
+    if from_date.len() < time_fmt.len() || to_date.len() < time_fmt.len() {
+        return Err(anyhow!("Invalid Date Input Format"));
+    }
+
+    let from_date_str: &str = &from_date[0..8];
+    let to_date_str: &str = &to_date[0..8];
+
+    let from_date_in = match NaiveDate::parse_from_str(from_date_str, "%Y%m%d") {
+        Ok(d) => d,
+        Err(_) => return Err(anyhow!("Invalid Date Input Format")),
```

Comment:
Suggest making the error message specific to the error such as "Invalid from date"

Reply:
Will do !